src/main/java/com/impactupgrade/nucleus/service/logic/AccountingService.java

```diff
@@ -47,55 +47,64 @@ public void processTransaction(PaymentGatewayEvent paymentGatewayEvent) throws E
+    protected void processDonation(CrmDonation crmDonation) throws Exception {
+        CrmAccount crmAccount = getDonationAccount(crmDonation);
         CrmContact crmContact = getDonationContact(crmDonation);
         if (crmContact == null) {
             // Should be unreachable
             env.logJobWarn("Failed to find crm contact for crm donation {}!", crmDonation.id);
             return;
         }
 
-        String contactId = _accountingPlatformService.get().updateOrCreateContact(crmContact);
+        //TODO: need to decide if we need to keep "account >> faux contact" concept here of move it to downstream
+        String contactId = _accountingPlatformService.get().updateOrCreateContact(crmAccount, Optional.of(crmContact));
         if (!Strings.isNullOrEmpty(contactId)) {
             env.logJobInfo("Upserted contact: {}", contactId);
 
-            AccountingTransaction accountingTransaction = toAccountingTransaction(crmDonation, contactId, crmContact.id);
+            AccountingTransaction accountingTransaction = toAccountingTransaction(crmDonation, contactId, crmAccount.id, crmContact.id);
             String transactionId = _accountingPlatformService.get().createTransaction(accountingTransaction);
             env.logJobInfo("Created transaction: {}", transactionId);
         }
     }
 
+    private CrmAccount getDonationAccount(CrmDonation crmDonation) throws Exception {
+        if (crmDonation.account == null || StringUtils.isEmpty(crmDonation.account.id)) {
+            return null;
+        }
+        // We retrieve the full objects from the CRM here,
+        // in case they have additional info over what was in the payment event.
+        return env.donationsCrmService().getAccountById(crmDonation.account.id).orElse(null);
+    }
+
     private CrmContact getDonationContact(CrmDonation crmDonation) throws Exception {
+        if (crmDonation.contact == null || StringUtils.isEmpty(crmDonation.contact.id)) {
+            return null;
+        }
         // We retrieve the full objects from the CRM here, in case they have additional info over what was in
         // the payment event.
-        CrmContact crmContact = null;
-        if (!StringUtils.isEmpty(crmDonation.account.id)) {
-            CrmAccount crmAccount = env.donationsCrmService().getAccountById(crmDonation.account.id).orElse(null);
-            // create a faux contact for org type account
-            if (crmAccount != null && crmAccount.recordType == EnvironmentConfig.AccountType.ORGANIZATION) {
-                crmContact = new CrmContact();
-                crmContact.id = crmAccount.id;
-                String[] firstLastName = Utils.fullNameToFirstLast(crmAccount.name);
-                crmContact.firstName = firstLastName[0];
-                crmContact.lastName = firstLastName[1];
-                crmContact.mailingAddress = crmAccount.billingAddress;
-                crmContact.crmRawObject = crmAccount.crmRawObject;
-            }
-        }
```

Comment:
Looking at this again and wondering if we already had what they were expecting, but I'm going to email Maria and double check. The above code was creating a contact using the biz name and address already. Which I THINK is all they were asking for...